site/cypress/e2e/Cypress-Feature/live_chat.spec.js

```diff
@@ -0,0 +1,419 @@
+        const isCurrentlyEnabled = $checkbox.is(':checked');
+        if (isCurrentlyEnabled !== enableChat) {
+            cy.wrap($checkbox).click();
+        }
+    });
+};
+
+const chatroomExists = (title) => {
+    return cy.get('body').then(($body) => {
+        if ($body.find('[data-testid="chatroom-item"]').length === 0) {
+            return false;
+        }
+
+        return cy.get('[data-testid="chatroom-item"]').then(($rows) => {
+            return $rows.filter((_, el) => {
+                return Cypress.$(el).find('[data-testid="chatroom-title"]').text().trim() === title;
+            }).length > 0;
+        });
+    });
+};
+
+const deleteChatroom = (title) => {
+    cy.reload();
+    chatroomExists(title).then((exists) => {
+        if (exists) {
+            endChatSession(title);
+            cy.window().then((win) => {
+                cy.stub(win, 'confirm').callsFake((confirmText) => {
+                    expect(confirmText).to.contain(`This will delete chatroom '${title}'. Are you sure?`);
+                    return true;
+                });
+            });
+
+            getChatroom(title).find('[data-testid="delete-chatroom"]').first().click();
+            deleteChatroom(title);
+        }
+    });
+};
+
+const checkChatExists = (title, exists = true) => {
+    if (exists) {
+        cy.get('[data-testid="chatroom-title"]')
+            .contains(title)
+            .should('exist');
+    }
+    else {
+        cy.get('body').then(($body) => {
+            if ($body.find('[data-testid="chatroom-item"]').length !== 0) {
+                cy.get('[data-testid="chatroom-title"]')
+                    .contains(title)
+                    .should('not.exist');
+            }
+        });
+    }
+};
+
+const checkDescription = (title, description) => {
+    getChatroom(title).find('[data-testid="chatroom-description"]').then(($el) => {
+        expect($el.text().trim()).to.equal(description);
+    });
+};
+
+const checkHost = (title, host) => {
+    getChatroom(title).find('[data-testid="chatroom-host"]').then(($el) => {
+        expect($el.text().trim()).to.equal(host);
+    });
+};
+
+const checkAnon = (title, expectedAnon) => {
+    const exist = expectedAnon ? 'exist' : 'not.exist';
+    getChatroom(title).then(($chatroom) => {
+        cy.wrap($chatroom).find('[data-testid="chat-join-btn"]').should('exist');
+        cy.wrap($chatroom).find('[data-testid="anon-chat-join-btn"]').should(exist);
+    });
+};
+
+const createChatroom = (title, description, isAnon) => {
+    cy.get('[data-testid="new-chatroom-btn"]').should('exist');
+    cy.get('[data-testid="new-chatroom-btn"]').click();
+    cy.get('[data-testid="chatroom-name-entry"]').should('exist').type(title, { force: true });
+    cy.get('[data-testid="chatroom-description-entry"]').should('exist').type(description, { force: true });
+    if (!isAnon) {
+        cy.get('[data-testid="enable-disable-anon"]').should('exist').click();
+    }
```

Comment:
.should('exist')... is redundant, because if it doesn't exist, then it will fail anyway, and if it does exist, it will pass without the .should('exist').